### PR TITLE
Make functional tests work on iOS too

### PR DESCRIPTION
**Pull request by guperrot**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Make functional tests work on iOS too

## Related PRs or issues

[AB#73032](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/73032)
[AB#72923](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/72923)

### CODE REVIEW

File: Tests/Microsoft.AppCenter.Test.Functional/Analytics/TrackEventTest.cs [hunk 1]
```diff
@@ -68,9 +72,9 @@ public async Task TrackEventWithPropertiesAsync()
 
             // Verify.
             Assert.Equal("POST", httpNetworkAdapter.Method);
-            var actualEventName = (string)httpNetworkAdapter.JsonContent["logs"][0]["name"];
+            var actualEventName = (string)httpNetworkAdapter.JsonContent.SelectTokens($"$.logs[?(@.type == 'event')]").ToList()[0]["name"];
```
Comment (guperrot) [inline Tests/Microsoft.AppCenter.Test.Functional/Analytics/TrackEventTest.cs]: in iOS, the start session log is usually in the same batch as the event.
on Android timing of the test is different and they are in 2 separate batches.

File: SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics/Analytics.cs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
```
Comment (jaelim-ms) [inline SDK/AppCenterAnalytics/Microsoft.AppCenter.Analytics/Analytics.cs]: Remove this one.

File: Tests/Microsoft.AppCenter.Test.Functional/Analytics/TrackEventTest.cs (hunk 1 above)
Comment (jaelim-ms) [reply Tests/Microsoft.AppCenter.Test.Functional/Analytics/TrackEventTest.cs]: This should be mentioned in the code. It is useful.

File: SDK/AppCenter/Microsoft.AppCenter.iOS.Bindings/ApiDefinition.cs [hunk 3]
```diff
@@ -278,4 +283,68 @@ interface MSWrapperLogger
...
+        [Export("setEnabled:")]
+        void SetEnabled(bool enabled);
+    }
+
+    // @protocol MSHttpClientDelegate <NSObject>
+    [Protocol]
+    [BaseType(typeof(NSObject))]
+    interface MSHttpClientDelegate
+    {
+        // -(void)willSendHTTPRequestToURL:(NSURL *)url withHeaders:(NSDictionary<NSString *, NSString *> *)headers;
```
Comment (jaelim-ms) [inline SDK/AppCenter/Microsoft.AppCenter.iOS.Bindings/ApiDefinition.cs]:
```suggestion
        // -(void)willSendHTTPRequestToURL:(NSURL *)url withHeaders:(nullable NSDictionary<NSString *, NSString *> *)headers;
```

**Pull request merged**